alacritty/src/event.rs

```diff
@@ -1024,13 +1068,26 @@ impl<N: Notify + OnResize> Processor<N> {
                     },
                     WindowEvent::Focused(is_focused) => {
                         if window_id == processor.ctx.window.window_id() {
-                            processor.ctx.terminal.is_focused = is_focused;
-                            processor.ctx.terminal.dirty = true;
+                            let terminal = &mut processor.ctx.terminal;
+                            terminal.is_focused = is_focused;
+                            terminal.dirty = true;
 
                             if is_focused {
                                 processor.ctx.window.set_urgent(false);
+
+                                // Reenable blinking cursor based on current mode.
+                                let config = &processor.ctx.config;
+                                let cursor_style = if terminal.mode().contains(TermMode::VI) {
+                                    let style = config.cursor.style();
+                                    config.cursor.vi_mode_style().unwrap_or(style)
+                                } else {
+                                    config.cursor.style()
+                                };
+                                processor.ctx.update_cursor_blinking(cursor_style.blinking);
                             } else {
                                 processor.ctx.window.set_mouse_visible(true);
+
+                                processor.ctx.update_cursor_blinking(false);
```

Comment:
You unconditionally enable blinking even if the blinking cursor mode was unset before, on focus change, this is noticeable in e.g. weechat which disables blinking completely, and after refocus the cursor starts blinking, which shouldn't be. You should track whether the current mode/cursor allows the cursor blinking in a first place, to handle unset of the mode properly.